uamqp/sender.py

```diff
@@ -149,10 +150,12 @@ def _send_unsent_messages(self):
         self._unsent_messages = unsent
 
     def send_transfer(self, message, **kwargs):
-        if self._is_closed:
-            raise ValueError("Link already closed.")
+        self._check_if_closed()
         if self.state != LinkState.ATTACHED:
-            raise ValueError("Link is not attached.")
+            raise AMQPConnectionError(
```

Comment:
Should this be a ConnectionError?

Reply:
ConnectionError doesn't exist anymore.. since it's a sender, I would go with an AMQPLinkError first, also left a TODO comment